K2fov/fov.py

```diff
@@ -207,6 +209,85 @@ def getBoresight(self):
     # Sky -> pixel code
     ###
 
+    def isOnSiliconList(self, ra_deg, dec_deg, padding_pix=DEFAULT_PADDING):
+        """similar to isOnSilicon() but takes lists as input"""
+        ch, col, row = self.getChannelColRowList(ra_deg, dec_deg)
+        out = np.zeros(len(ch), dtype=bool)
+        for channel in set(ch):
+            mask = (ch == channel)
+            if channel in self.brokenChannels:
+                continue
+            if channel > 84:
+                continue
+            out[mask] = self.colRowIsOnSciencePixelList(col[mask], row[mask], padding_pix)
+        return out
+        
+    def getChannelColRowList(self, ra, dec, wantZeroOffset=False,
+                         allowIllegalReturnValues=True):
+        """similar to getChannelColRow() but takes lists as input"""
+        try:
+            ch = self.pickAChannelList(ra, dec)
+        except ValueError:
+            logger.warning("WARN: %.7f %.7f not on any channel" % (ra, dec))
+            return (0, 0, 0)
+        
+        col = np.zeros(len(ch))
+        row = np.zeros(len(ch))
+        for channel in set(ch):
+            mask = (ch == channel)
+            col[mask], row[mask] = self.getColRowWithinChannelList(ra[mask], dec[mask], channel, 
+                                                        wantZeroOffset, allowIllegalReturnValues)
+        return (ch, col, row)        
+
+    def pickAChannelList(self, ra_deg, dec_deg):
+        """similar to pickAChannel() but takes lists as input"""
+        cRa = self.currentRaDec[:, 3]  # Ra of each channel corner
+        cDec = self.currentRaDec[:, 4]  # dec of each channel corner
+        catalog = SkyCoord(cRa*u.deg, cDec*u.deg)
+        position = SkyCoord(ra_deg*u.deg, dec_deg*u.deg)
+        idx, _, _ = position.match_to_catalog_sky(catalog)
+        return self.currentRaDec[idx, 2]
+
+    def getColRowWithinChannelList(self, ra, dec, ch, wantZeroOffset=False,
+                               allowIllegalReturnValues=True):
+        """similar to getColRowWithinChannel() but takes lists as input"""
+        x, y = self.defaultMap.skyToPix(ra, dec)
+        kepModule = self.getChannelAsPolygon(ch)
+        r = np.array(zip(x, y)) - kepModule.polygon[0, :]
```

Comment:
Looks like this needs to be `np.array(list(zip(x, y)))` instead of `np.array(zip(x, y))`, because `zip()` in Python 3 returns an iterator rather than a list.